agent/logging.go

```diff
@@ -116,6 +118,8 @@ func (a *Agent) prepLogger(tc *taskContext, c *model.LoggerConfig, commandName s
 		SendToGlobalSender: a.opts.SendTaskLogsToGlobalSender,
 		AWSCredentials:     pail.CreateAWSCredentials(tc.taskConfig.TaskSync.Key, tc.taskConfig.TaskSync.Secret, ""),
 	}
+	config.Expansions = tc.taskConfig.NewExpansions
+	config.ExpansionsToRedact = redactList(tc.taskConfig.ProjectVars, tc.taskConfig.Redacted)
```

Comment:
We should probably also check during expansions.write and expansions.update if the new expansion needs to be redacted?

Reply:
Expansions to redact are already checked in `expansions.write`.

I left the work to redact new private expansions for [DEVPROD-4107](https://jira.mongodb.org/browse/DEVPROD-4107); I imagine that it will be much cleaner to implement that functionality together.